gwnr/waveform/utils.py

```diff
@@ -259,3 +259,88 @@ def get_isco_frequency(mass1, mass2, spin1z, spin2z):
     f_isco = m_omg_isco / (lal.PI * (mass1 + mass2) * lal.MTSUN_SI)
     print(f"Value of f at ISCO: {f_isco}Hz")
     return f_isco
+
+
+def f_ISCO_spin(mass1, mass2, spin1z, spin2z):
+    """
+    Kerr ISCO frequency fitting formula for aligned-spins binary black holes
+
+    Parameters:
+    ----------
+    mass1, mass2   -- Binary's component masses (in solar masses)
+    spin1z, spin2z   -- z-components of component dimensionless spins (lies in [0,1))
+
+    Returns: Kerr ISCO frequency (in Hz)
+    -------
+    """
+    G = 6.6742 * 1e-11
+    cc = 2.99792458 * 1e8
+    Msun = (G / cc**3) * 1.9891 * 1e30
+    k00 = -3.821158961
+    k01 = -1.2019
+    k02 = -1.20764
+    k10 = 3.79245
+    k11 = 1.18385
+    k12 = 4.90494
+    Zeta = 0.41616
+
+    m1 = mass1 * Msun
+    m2 = mass2 * Msun
+    M = m1 + m2
+    eta = (m1 * m2) / (M**2)
+    z = 0
+
+    atot = (spin1z + spin2z * (m2 / m1) ** 2) / ((1 + (m2 / m1)) ** 2)
+    aeff = atot + Zeta * eta * (spin1z + spin2z)
+
+    Z1 = 1 + ((1 - aeff**2) ** (1 / 3)) * (
+        ((1 + aeff) ** (1 / 3)) + ((1 - aeff) ** (1 / 3))
+    )
+    Z2 = np.sqrt(3 * aeff**2 + Z1**2)
+    # riscocap = 3 + Z2 - ((aeff) / abs(aeff)) * np.sqrt((3 - Z1) * (3 + Z1 + 2 * Z2))
+    riscocap = 3 + Z2 - np.sign(aeff) * np.sqrt((3 - Z1) * (3 + Z1 + 2 * Z2))
+    # Equivalent to the above commented expression, and works also for aeff=0.
```

Comment:
Is this so because `Z1` goes to `3` as `aeff --> 0` , and so the factor proportional to `(3 - Z1)` becomes zero anyway..?

Reply:
Yes, exactly.